src/Processors/QueryPlan/ReadFromMergeTree.h

```diff
@@ -252,7 +252,7 @@ class ReadFromMergeTree final : public SourceStepWithFilter
     Pipe read(RangesInDataParts parts_with_range, Names required_columns, ReadType read_type, size_t max_streams, size_t min_marks_for_concurrent_read, bool use_uncompressed_cache);
     Pipe readFromPool(RangesInDataParts parts_with_range, Names required_columns, PoolSettings pool_settings);
     Pipe readFromPoolParallelReplicas(RangesInDataParts parts_with_range, Names required_columns, PoolSettings pool_settings);
-    Pipe readInOrder(RangesInDataParts parts_with_ranges, Names required_columns, PoolSettings pool_settings, ReadType read_type, UInt64 limit);
+    Pipe readInOrder(RangesInDataParts parts_with_ranges, Names required_columns, PoolSettings pool_settings, ReadType read_type, UInt64 limit, bool need_virtual_row = false);
```

Comment:
Let's remove default value for need_virtual_row flag.

We need some similar flag to `requestReadingInOrder` function (and probably to InputOrderInfo structure).
We also need to check that all the call sides can support virtual row.